exonum-java-binding-core/rust/src/runtime/cmd.rs

```diff
@@ -138,3 +132,67 @@ impl CommandExtension for Finalize {
         Ok(context)
     }
 }
+
+pub struct Run;
+
+impl CommandExtension for Run {
+    fn args(&self) -> Vec<Argument> {
+        vec![
+            Argument::new_named(
+                EJB_JVM_ARGUMENTS,
+                false,
+                "Additional parameters for JVM. Must not have a leading dash. \
+                 For example, `Xmx2G` or `Xdebug`",
+                None,
+                "ejb-jvm-args",
+                true,
+            ),
+            Argument::new_named(
+                EJB_JVM_DEBUG_SOCKET,
+                false,
+                "Allows JVM being remotely debugged. Takes a socket address as a parameter in form \
+                of `HOSTNAME:PORT`. Must not have a leading dash. For example, `localhost:8000`",
+                None,
+                "jvm-debug",
+                false,
+            ),
+        ]
+    }
+
+    fn execute(&self, mut context: Context) -> Result<Context, failure::Error> {
```

Comment:
I don't understand what happens here. Would you please clarify the intention of this code?

Reply:
It creates a new Config structure mostly by copying content from the old one because here we can fill two other fields that were not accessible at previous steps. Well, this code part is also under question for me because I was wondering if it worth extracting those fields out of this structure. And most probably, better to rewrite and get rid of those tricks with initialization of `user_parameters` and `jvm_debug_socket` on the configuration step.

Reply:
I see, thanks, I'd find some ways to make this intention more apparent (comments clarifying individual steps, extracting some of these steps into separate functions, etc).

Reply:
I still find it quite hard to grasp.